csharp/src/Glacier2/SessionFactoryHelper.cs

```diff
@@ -19,180 +19,234 @@ namespace ZeroC.Glacier2
     /// session factory to create another connection.</summary>
     public class SessionFactoryHelper
     {
-        /// <summary>Creates a SessionFactory object.</summary>
-        /// <param name="callback">The callback for notifications about session establishment.</param>
-        /// <param name="properties">Optional properties used for communicator initialization.</param>
-        /// <param name="logger">Optional logger used for communicator initialization.</param>
-        /// <param name="observer">Optional communicator observer used for communicator initialization.</param>
-        public SessionFactoryHelper(ISessionCallback callback,
-                             Dictionary<string, string> properties,
-                             ILogger? logger = null,
-                             ICommunicatorObserver? observer = null)
-        {
-            _callback = callback;
-            _properties = properties;
-            _logger = logger;
-            _observer = observer;
-
-            SetDefaultProperties();
-        }
-
-        /// <summary>Set the router object identity.</summary>
-        /// <param name="identity">The Glacier2 router's identity.</param>
-        public void SetRouterIdentity(Identity identity)
+        /// <summary>Gets or sets the request context to use while establishing a connection to the Glacier2 router.
```

Comment:
```suggestion
        /// <summary>The request context to use while establishing a connection to the Glacier2 router.
```
I'm not sure what our typical pattern is, but I don't see the need to have "Gets or sets" for properties with getters and setters.

Reply:
Regarding "Gets or sets" I think this is the preferred way for C# properties documentation

see https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1623.md